src/StoreKeeper/WooCommerce/B2C/Core.php

```diff
@@ -584,47 +574,53 @@ public function syncAllPaidAndProcessingOrders(): void
+        wp_localize_script('image-upload-script', 'ajax_object', [
             'ajax_url' => admin_url('admin-ajax.php'),
             'nonce' => wp_create_nonce('upload_product_image_nonce'),
-            'translations' => array(
-                'please_select_image' => __('Please select an image first.', I18N::DOMAIN),
-                'image_too_small' => __('Image is too small. Minimum dimensions are {width}x{height}.', I18N::DOMAIN),
-                'image_too_large' => __('Image is too large. Maximum dimensions are {width}x{height}.', I18N::DOMAIN),
-                'upload_success' => __('Image uploaded successfully!', I18N::DOMAIN),
-                'upload_error' => __('An error occurred while uploading the image.', I18N::DOMAIN),
-            ),
-        ));
+            'translations' => [
+                'please_select_image' => esc_html__('Please select an image first.', I18N::DOMAIN),
+                'image_too_small' => esc_html__('Image is too small. Minimum dimensions are {width}x{height}.', I18N::DOMAIN),
+                'image_too_large' => esc_html__('Image is too large. Maximum dimensions are {width}x{height}.', I18N::DOMAIN),
+                'upload_success' => esc_html__('Image uploaded successfully!', I18N::DOMAIN),
+                'upload_error' => esc_html__('An error occurred while uploading the image.', I18N::DOMAIN),
+            ],
+        ]);
     }
 
     public function handleProductImageUpload()
     {
         if (!isset($_POST['nonce']) || !wp_verify_nonce($_POST['nonce'], 'upload_product_image_nonce')) {
-            wp_send_json_error(['message' => __('Nonce verification failed', I18N::DOMAIN)]);
+            wp_send_json_error(['message' => esc_html__('Nonce verification failed', I18N::DOMAIN)]);
             return;
         }
-        
+
         if (isset($_FILES['file']) && !empty($_FILES['file']['tmp_name'])) {
-            $uploaded_file = $_FILES['file'];
+            $uploadedFile = $_FILES['file'];
 
-            $upload_dir = wp_upload_dir();
-            $upload_path = $upload_dir['path'] . '/' . basename($uploaded_file['name']);
+            $uploadDir = wp_upload_dir();
+            $uploadPath = $uploadDir['path'].'/'.basename($uploadedFile['name']);
+            $allowedMimeTypes = ['image/jpeg', 'image/jpg', 'image/png'];
+            $fileMimeType = mime_content_type($uploadedFile['tmp_name']);
 
-            if (move_uploaded_file($uploaded_file['tmp_name'], $upload_path)) {
-                $image_url = $upload_dir['url'] . '/' . basename($uploaded_file['name']);
+            if (!in_array($fileMimeType, $allowedMimeTypes)) {
+                wp_redirect(add_query_arg('upload_error', 'invalid_file_type', wp_get_referer()));
+                exit;
+            }
+
+            if (move_uploaded_file($uploadedFile['tmp_name'], $uploadPath)) {
+                $image_url = $uploadDir['url'].'/'.basename($uploadedFile['name']);
```

Comment:
This is an ajax action, wp_redirect is not appropriate
```suggestion
            if (!in_array($fileMimeType, $allowedMimeTypes)) {
                wp_send_json_error(['message' => esc_html__('Invalid file type', I18N::DOMAIN)]);
            } else if (move_uploaded_file($uploadedFile['tmp_name'], $uploadPath)) {
                $image_url = $uploadDir['url'].'/'.basename($uploadedFile['name']);
```